driver-core/src/test/functional/com/mongodb/client/model/expressions/DateExpressionsFunctionalTest.java

```diff
@@ -0,0 +1,148 @@
+
+    private final Instant instant = Instant.parse("2007-12-03T10:15:30.005Z");
+
+    @Test
+    public void literalsTest() {
+        assertExpression(
+                instant,
+                of(instant),
+                "{'$date': '2007-12-03T10:15:30.005Z'}");
+    }
+
+    @Test
+    public void yearTest() {
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/year/
+        assertExpression(
+                2007,
+                of(instant).year(),
+                "{'$year': {'$date': '2007-12-03T10:15:30.005Z'}}");
+    }
+
+    @Test
+    public void monthTest() {
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/month/
+        assertExpression(
+                12,
+                of(instant).month(),
+                "{'$month': {'$date': '2007-12-03T10:15:30.005Z'}}");
+    }
+
+    @Test
+    public void dayOfMonthTest() {
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/dayOfMonth/
+        assertExpression(
+                3,
+                of(instant).dayOfMonth(),
+                "{'$dayOfMonth': {'$date': '2007-12-03T10:15:30.005Z'}}");
+    }
+
+    @Test
+    public void dayOfWeekTest() {
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/dayOfWeek/
+        assertExpression(
+                2,
+                of(instant).dayOfWeek(),
+                "{'$dayOfWeek': {'$date': '2007-12-03T10:15:30.005Z'}}");
+    }
+
+    @Test
+    public void dayOfYearTest() {
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/dayOfYear/
+        assertExpression(
+                337,
+                of(instant).dayOfYear(),
+                "{'$dayOfYear': {'$date': '2007-12-03T10:15:30.005Z'}}");
+    }
+
+    @Test
+    public void hourTest() {
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/hour/
+        assertExpression(
+                10,
+                of(instant).hour(),
+                "{'$hour': {'$date': '2007-12-03T10:15:30.005Z'}}");
+    }
+
+    @Test
+    public void minuteTest() {
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/minute/
+        assertExpression(
+                15,
+                of(instant).minute(),
+                "{'$minute': {'$date': '2007-12-03T10:15:30.005Z'}}");
+    }
+
+    @Test
+    public void secondTest() {
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/second/
+        assertExpression(
+                30,
+                of(instant).second(),
+                "{'$second': {'$date': '2007-12-03T10:15:30.005Z'}}");
+    }
+
+    @Test
+    public void weekTest() {
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/week/
+        assertExpression(
+                48,
+                of(instant).week(),
+                "{'$week': {'$date': '2007-12-03T10:15:30.005Z'}}");
+    }
```

Comment:
This is the only test where the expected value can't be trivially computed with the Java SE API.